game/script-topics.rpy

```diff
@@ -599,6 +599,573 @@ label talk_eating_well:
+
+        if player_input_count >= 3:
+            # Escape route if we somehow get stuck
+            n 1kllaj "I...{w=0.3} don't think we're getting anywhere with this,{w=0.1} [player]."
+            n 1nnmss "Maybe we should just try again later?"
+            return {"lock" : None}
+
+        else:
+            $ player_input = renpy.input("Enter your API key (or type Nevermind to go back):")
+
+        if not player_input or player_input == "":
+            n 1nlraj "Uh...{w=0.3} Did you actually type anything,{w=0.1} [player]?"
+            n 1nwmss "I didn't get that...{w=0.3} can you try again for me?"
+            $ player_input_count += 1
+
+        elif player_input.replace(" ", "").lower() == "nevermind":
+            # Allow the player to back out
+            n 1knmaj"Huh?{w=0.2} You don't wanna continue?"
+            n 1nwmpu "That's fine,{w=0.1} I guess."
+            n 1nnmss "Just let me know when you're ready,{w=0.1} 'kay?"
+            return {"lock" : None}
+
+        else:
+            # Get ready to lead in to the next stage of setup
+            $ player_input_valid = True # Kill the loop!
+            $ persistent.weather_api_key = player_input
+            n 1nnmlg "Alright!{w=0.2} I got it!"
+            n 1nlrbg "Let me just work all of this out...{w=0.3} I might be a few minutes,{w=0.1} just so you know."
+
+            # Final dialogue permutations for that personal touch
+            if jn_affinity.get_affinity_state() >= store.jn_affinity.ENAMORED:
+                $ chosen_descriptor = random.choice(jn_globals.DEFAULT_PLAYER_DESCRIPTORS)
+                n 1nchbg "But thanks,{w=0.1} [player]!{w=0.2} You're [chosen_descriptor],{w=0.1} you know that?"
+            else:
+                n 1nchbg "But thanks for all your help, [player]!"
+
+            if player_was_rude:
+                n 1nwrts "Even despite your sourness earlier.{w=0.2} Ehehe."
+
+            elif player_was_quiet:
+                n 1nllss "Even if you were a little hesitant at first.{w=0.2} Ehehe."
+
+            if jn_affinity.get_affinity_state() >= store.jn_affinity.LOVE:
+                $ chosen_endearment = random.choice(jn_globals.DEFAULT_PLAYER_ENDEARMENTS)
+                n 1nchbg "Love you,{w=0.1} [chosen_endearment]!"
+            # It can take a few hours for the API key to activate
+            # so we wait ~2 hours before checking if the key is working
+            $ persistent.weather_validate_apikey_in_time = datetime.datetime.now()+datetime.timedelta(minutes=0.5)
+
+    return {"lock!" : None}
+
+init 5 python:
+    registerTopic(
+        Topic(
+            persistent._topic_database,
+            label="talk_weather_setup_part2",
+            unlocked=False,
+            nat_says=True,
+            category=["weather"]
+        ),
+        topic_group=TOPIC_TYPE_NORMAL
+    )
+
+label talk_weather_setup_part2:
+
+    # API key isn't valid; we prompt the player for another
+    if not weather.is_api_key_valid():
```

Comment:
I'm getting an odd exception thrown on this topic; it looks like the API call is failing due to a URL error on the urllib2 library?
![image](https://user-images.githubusercontent.com/57731669/145099405-3be692a7-00e8-4cce-bd84-8273d448b711.png)